### PR TITLE
add survey.js@0.12.1 w/ npm auto-update and manually add old versions

### LINKED ISSUE #9771: [Request] Add survey.js
**Library name:** survey.js
**Git repository url:** http://github.com/andrewtelnov/surveyjs
**npm package url(optional):** survey-knockout, survey-react
**License(s):** MIT
**Official homepage:** http://surveyjs.org/
**Wanna say something? Leave message here:** Very useful in creating client-side surveys. Although the repo has only 80 stars, combined npm packages downloads per month are well over 500 (295  + 308). There are two npm packages, as survey.js supports two frameworks.

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #9771 
Related issue(s): # #
@kennynaoh please help me review this PR, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): http://github.com/andrewtelnov/surveyjs
 * Official website (optional, not the repository): http://surveyjs.org/
 * NPM package url (optional): https://www.npmjs.com/package/survey-knockout
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 101
   - Count of watchers: 18
   - Count of forks: 48
 * NPM download stats (optional):
   - Downloads in the last day: 4
   - Downloads in the last week: 172
   - Downloads in the last month: 700

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/survey.js/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "author": {
+    "name": "andrewtelnov",
+    "email": "andrew.telnov@gmail.com"
+  },
+  "npmName": "survey-knockout",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "survey.ko*"
```
Comment (PeterDaveHello) [inline ajax/libs/survey.js/package.json]: Let's use `"**/survey.ko*"`

Comment (sufuf3) [inline ajax/libs/survey.js/package.json]: I modified it. @kennynaoh please help me review this PR, thank you.

File: ajax/libs/survey.js/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "description": "survey.js is a JavaScript Survey Library. It is a modern way to add a survey to your website. It uses JSON for survey metadata and results.",
+  "keywords": [
+    "Survey",
+    "JavaScript",
+    "Bootstrap",
+    "Library",
+    "Knockout"
+  ],
+  "filename": "survey.ko.min.js",
+  "homepage": "https://surveyjs.org/",
```
Comment (kennynaoh) [inline ajax/libs/survey.js/package.json]: Let's use `http` because the website doesn't support `https`

Comment (sufuf3) [inline ajax/libs/survey.js/package.json]: I modified it. @kennynaoh please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
#9771 also mentions survey-react, but not only survey-knockout

**Comment by sufuf3** [comment]
@kennynaoh please help me review this PR, thank you.

**Comment by sufuf3** [comment]
@kennynaoh please help me review this PR again, thank you.

**Comment by sufuf3** [comment]
ping @kennynaoh

**Comment by kennynaoh** [comment]
I think we should send four PR in this case instead of four commits, should we? @PeterDaveHello

**Comment by PeterDaveHello** [comment]
It's fine.

**Comment by sufuf3** [comment]
@kennynaoh should I just rebase master, then it well be OK?

**Comment by kennynaoh** [comment]
@sufuf3 Please minify `survey.css`.

**Comment by sufuf3** [comment]
@kennynaoh please help me review this PR again, thank you.

**Comment by sufuf3** [comment]
ping @kennynaoh

**Comment by sufuf3** [comment]
@kennynaoh I fixed the structure of these old versions, added the new lib survey-vue, and updated the new version.
Please help me review this PR again, thank you.

@cdnjs/intern3 please help me review this PR, thank you.

**Comment by kennynaoh** [comment]
There is only css file in the v0.11.0 of survey-knockout, could you confirm with author? Thanks!

**Comment by sufuf3** [comment]
@kennynaoh I added the v0.11.0 of survey-knockout files from https://github.com/surveyjs/surveyjs/releases/tag/0.11.
Please help me review this PR again, thank you.
@cdnjs/intern3 please help me review this PR, thank you.

**Comment by kennynaoh** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
@extend1994, @hare1039, and @clairetsai818 ?

File: ajax/libs/survey.js/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/andrewtelnov/surveyjs.git"
+  },
+  "author": {
+    "name": "andrewtelnov",
+    "email": "andrew.telnov@gmail.com"
+  },
+  "npmName": "survey-knockout",
```
Comment (pvnr0082t) [inline ajax/libs/survey.js/package.json]: `survey.js` also fetch the files from `survey-knockout`?

**Comment by sufuf3** [comment]
@pvnr0082t @extend1994 sorry about that I shouldn't host the `survey.js`.
It's not a lib form git repo or npm, so I remove it.
Please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
I think we had better not close the issue by one of this commit, you can close it manually  after this PR being merged.

**Comment by sufuf3** [comment]
@maruilian11 you meaning that I should change the commit message?
from 
```
add survey-vue@0.12.1 w/ npm auto-update  
close #9771, cc @andrewtelnov
```
to
```
Add survey-vue@0.12.1 w/ npm auto-update  
cc #9771, cc @andrewtelnov
```

**Comment by sufuf3** [comment]
ping @maruilian11 @pvnr0082t

**Comment by PeterDaveHello** [comment]
I got strange situation here, the branch I fetched locally has different orders, not like what do I see on GitHub.

@sufuf3 which one is correct? I mean the same as your local.

What I see on GitHub:
image

On my local:


(PS: the order should be reversed.)

**Comment by maruilian11** [comment]
@sufuf3 yes, please.

**Comment by sufuf3** [comment]
@PeterDaveHello on my local is same as yours.
I don't know how to change the commit orders on GitHub.

**Comment by PeterDaveHello** [comment]
@sufuf3 Same order and same hashes?

**Comment by sufuf3** [comment]
Yes
image

**Comment by sufuf3** [comment]
@PeterDaveHello  Yes
image

**Comment by PeterDaveHello** [comment]
@sufuf3 Then we should report a bug to GitHub.

**Comment by sufuf3** [comment]
@maruilian11 I changed the commit message.
Please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
please update the title

**Comment by sufuf3** [comment]
@maruilian11 Is the new title OK?
Because this PR include many libraries from npm, and not all the libraries manually add the same old versions.
So I use `add survey.js@0.12.1 w/ npm auto-update and manually add old versions` as this title.

If there is any suggest, please let me know.
Thank you.

**Comment by maruilian11** [comment]
@sufuf3 I think this PR is good now, please rebase the branch with our latest master.

**Comment by sufuf3** [comment]
@maruilian11 I rebased the branch with the latest master.
Please help me review this PR again, thank you.

**Pull request merged**